### ISSUE TITLE
Views - hierarchy - Error

### ISSUE THREAD

**Issue by Scoppettuolo** [backdrop/backdrop-issues#2055]
Notice: Undefined property: TaxonomyVocabulary::$vid en views_handler_filter_term_node_tid->value_form() (línea 129 de E:\Mi PC\Disco local\Archivos de programa\Programas\Programas de sistema\Internet\XAMPP\XAMPP\htdocs\backdrop\core\modules\taxonomy\views\views_handler_filter_term_node_tid.inc).

I have try construct my view with...:

- Views - Has taxonomy term (with depth)
- Dropdown - Show hierarchy in dropdown

Note: * How Autocomplete the tems work. *

...but I can´t get that this work.

In Drupal 7 my Views work, but not in Backdrop.

Is my error, or the Backdrop?

**Comment by jenlampton** [comment]
In Backdrop Vocabularies are no longer objects (IIRC) so you'll likely need to change $vocab->vid to $vocab['vid'].  

Can you please check line 129 of backdrop\core\modules\taxonomy\views\views_handler_filter_term_node_tid.inc and see if that is the problem?

**Comment by Scoppettuolo** [comment]
Sorry for the late.

Change that no work for me, and the error is the same.

The terms disappear with the Hierarchy Dropdown.

**Comment by jenlampton** [comment]
Is the line number the same or is it telling you to change another instance? I expect there are likely a few of those and you may need to change them all.

**Comment by quicksketch** [comment]
The problem is `$vocabulary->vid` doesn't exist any more because we no longer have numeric IDs for taxonomy vocabularies. As part of the configuration management changes, vocabularies are stored in configuration files and so have machine names instead of a database auto-increment column.

Basically, instead of `$vocabulary->vid`, we should use `$vocabulary->machine_name` now. I'll look into seeing if I can reproduce this.

**Comment by Scoppettuolo** [comment]
That is the solution!

`$vocabulary->machine_name` works perfectly.

**Comment by klonos** [comment]
OK, steps to reproduce:

1. create a view of content
2. add "Content: Tags" as a filter criterion
3. select "dropdown" as selection type
4. select the "Show hierarchy in dropdown" option
5. apply
6. click the just-inserted "Content: Tags (or Unknown)" filter criterion in order to configure it (not the "settings" link next to it):

```
Notice: Undefined property: TaxonomyVocabulary::$vid in views_handler_filter_term_node_tid->value_form() (line 129 of C:\www\core\modules\taxonomy\views\views_handler_filter_term_node_tid.inc).
```

7. close the configure dialog and click the "settings" link of the filter criterion and either change its selection type to autocomplete or deselect the "Show hierarchy in dropdown" option.
8. apply
9. click the "Content: Tags (or Unknown)" filter criterion again...

No error this time.

**Comment by klonos** [comment]
Changing `$vocabulary->vid` to `$vocabulary->machine_name` in https://github.com/backdrop/backdrop/blob/1.x/core/modules/taxonomy/views/views_handler_filter_term_node_tid.inc#L129 solves the issue, but I see that this line is already corrected in 1.x ???

**Comment by quicksketch** [comment]
> but I see that this line is already corrected in 1.x ???

Oops, I accidentally changed that line in 80cbef0f5e0aea93aea76cd7b001870878c9b6ce while committing some D7 ports. However, that line is *not* yet fixed in 1.4.x. I'll file a PR to fix that branch as well. Updating that line again just to get the history right might be order as well.

**Comment by quicksketch** [comment]
Fixed in https://github.com/backdrop/backdrop/issues/1488. Thanks @Scoppettuolo for this report and testing the fix!
